src/react-components/loader.js

```diff
@@ -51,12 +52,14 @@ class Loader extends Component {
 
     if (!this.doneWithInitialLoad && this.loadingTimeout) window.clearTimeout(this.loadingTimeout);
 
-    this.loadingTimeout = window.setTimeout(() => {
-      this.doneWithInitialLoad = true;
-      if (this.props.onLoaded) {
-        this.props.onLoaded();
-      }
-    }, 1500);
+    if (this.props.environmentSceneLoaded) {
```

Comment:
does `onObjectLoaded` happen when the environment loads? if not, this change would imply this `if` could never be entered in the case where the environment loads last (but i'm assuming this was the main breaking case)

Reply:
Oh, I hadn't consider that. 
I updated the PR such that the loading screen will definitely finish no matter the order of the two relevant event handlers (`onObjectLoaded` in the `Loader` and `onFirstEnvironmentLoad` in hub.js)